clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/manifest/KubernetesUndoRolloutManifestOperation.java

```diff
@@ -61,11 +61,29 @@ public Void operate(List priorOutputs) {
 
     CanUndoRollout canUndoRollout = (CanUndoRollout) deployer;
 
+    Integer revision = description.getRevision();
+    if (description.getNumRevisionsBack() != null) {
+      getTask().updateStatus(OP_NAME, "Looking up rollout history...");
+      List<Integer> revisions = canUndoRollout.historyRollout(credentials,
+          coordinates.getNamespace(),
+          coordinates.getName());
+
+      revisions.sort(Integer::compareTo);
+      int numRevisions = revisions.size();
+      int targetRevisionIndex = numRevisions - description.getNumRevisionsBack() - 1;
+      if (targetRevisionIndex < 0) {
+        throw new IllegalArgumentException("There are " + numRevisions + " in total, cannot rollback " + description.getNumRevisionsBack());
```

Comment:
This message might be confusing out of context - could add "revision" after `numRevisions`.